src/software/simulation/er_force_simulator_robot.h

```diff
@@ -0,0 +1,285 @@
+     *
+     * @return the y-velocity of the robot, in global field coordinates, in m/s
+     */
+    float getVelocityY() override;
+
+    /**
+     * Returns the angular velocity of the robot, in rad/s
+     *
+     * @return the angular of the robot, in rad/s
+     */
+    float getVelocityAngular() override;
+
+    /**
+     * Returns the battery voltage, in volts
+     *
+     * @return the battery voltage, in volts
+     */
+    float getBatteryVoltage() override;
+
+    /**
+     * Fires the kicker, kicking the ball in the direction the robot is facing
+     * at the given speed if the ball is very close to the kicker
+     *
+     * @param speed_m_per_s How fast to kick the ball, in meters per second
+     */
+    void kick(float speed_m_per_s) override;
+
+    /**
+     * Fires the chipper, chipping the ball in the direction the robot is facing
+     * for the given distance if the ball is very close to the chipper
+     *
+     * @param speed_m_per_s How far to chip the ball (the distance to the first bounce)
+     * in meters
+     */
+    void chip(float distance_m) override;
+
+    /**
+     * Enables autokick on the robot. If the ball touches the kicker, the robot will
+     * kick the ball with the given speed.
+     *
+     * @param speed_m_per_s How fast to kick the ball in meters per second when
+     * the kicker is fired
+     */
+    void enableAutokick(float speed_m_per_s) override;
+
+    /**
+     * Enables autochip on the robot. If the ball touches the chipper, the robot will
+     * chip the ball the given distance.
+     *
+     * @param speed_m_per_s How far to chip the ball (distance to the first bounce)
+     * when the chipper is fired
+     */
+    void enableAutochip(float distance_m) override;
+
+    /**
+     * Disables autokick
+     */
+    void disableAutokick() override;
+
+    /**
+     * Disables autochip
+     */
+    void disableAutochip() override;
+
+    /**
+     * Returns true if autokick is enabled and false otherwise
+     *
+     * @return true if autokick is enabled and false otherwise
+     */
+    bool isAutokickEnabled() override;
+
+    /**
+     * Returns true if autochip is enabled and false otherwise
+     *
+     * @return true if autochip is enabled and false otherwise
+     */
+    bool isAutochipEnabled() override;
+
+    /**
+     * Sets the speed of the dribbler
+     *
+     * @param rpm The rpm to set for the dribbler
+     */
+    void setDribblerSpeed(uint32_t rpm) override;
+
+    /**
+     * Makes the dribbler coast until another operation is applied to it
+     */
+    void dribblerCoast() override;
+
+    /**
+     * Returns the temperature of the dribbler, in degrees C
+     *
+     * @return the temperature of the dribbler, in degrees C
+     */
+    unsigned int getDribblerTemperatureDegC() override;
+
+    /**
+     * Sets the target RPM of the wheel
+     *
+     * @param rpm the target RPM of the wheel
+     */
+    void setTargetRPMFrontLeft(float rpm);
```

Comment:
by our naming convention, I think we treat RPM as Rpm

Reply:
fixed